nmpolicy/internal/capture/capture_entry.go

```diff
@@ -16,13 +16,56 @@
 
 package capture
 
-import "github.com/nmstate/nmpolicy/nmpolicy/types"
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/lexer"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/parser"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/resolver"
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
 
 type CaptureEntry struct {
-	CaptureCache map[string]types.CaptureState
+	capturedStates map[string]map[string]interface{}
+	lexer          Lexer
+	parser         Parser
+	resolver       Resolver
+}
+
+func NewCaptureEntryWithLexerParserResolver(capturedStates map[string]types.CaptureState,
```

Comment:
Why do we need this exported method. The test is in the same package, cannot it just updated `CaptureEntry.lexer/parser/resolver`?

Reply:
The tests are at capure_test package.